lib/symboldatabase.cpp

```diff
@@ -3734,9 +3734,13 @@ bool Variable::arrayDimensions(const Settings& settings, bool& isContainer)
             // TODO: only perform when ValueFlow is enabled
             // TODO: collect timing information for this call?
             ValueFlow::valueFlowConstantFoldAST(const_cast<Token *>(dimension_.tok), settings);
-            if (dimension_.tok && (dimension_.tok->hasKnownIntValue() ||
-                                   (dimension_.tok->isTemplateArg() && !dimension_.tok->values().empty()))) {
-                dimension_.num = dimension_.tok->getKnownIntValue();
+            if (dimension_.tok) {
+                if (dimension_.tok->hasKnownIntValue())
+                    dimension_.num = dimension_.tok->getKnownIntValue();
+                else if (dimension_.tok->isTemplateArg() && !dimension_.tok->values().empty()) {
+                    assert(dimension_.tok->values().front().isKnown());
+                    dimension_.num = dimension_.tok->values().front().intvalue;
+                }
```

Comment:
@chrchr-github Please have a look. You changed this logic a few days ago in 46781df9657bfd31922e635466c3951af96f6a85.

Reply:
For template arguments, the values are only possible, not known. But since `getKnownIntValue()` just returns the first one, it is what we want in this case (I think).

Reply:
So like the previous behavior? Just unconditionally use the first value and set `dimension_.known` to `true`?

Reply:
Yes, pretty much. I imagine/hope that template arguments can have at most one value.

Reply:
Maybe we should set those values to known in the future, and prevent unwanted `knownConditionTrueFalse` warnings in a different way.

Reply:
> I imagine/hope that template arguments can have at most one value.

I will just add a different assert.